js/content/store.js

```diff
@@ -3119,9 +3174,138 @@ let WishlistPageClass = (function(){
                 DynamicStore.clear();
                 location.reload();
             });
-        }, true)
+        }, true);
     }
 
+    WishlistPageClass.prototype.showModalDialog = function() {
+        // Partly copied from shared_global.js
+        ExtensionLayer.runInPageContext(`function() {
+            let deferred = new jQuery.Deferred();
+            let fnOK = () => deferred.resolve();
+
+            let Modal = _BuildDialog(
+                "${Localization.str.export_wishlist}",
+                \`${this.exportModalTemplate}\`,
+                [], fnOK);
+            deferred.always(() => Modal.Dismiss());
+
+            Modal.m_fnBackgroundClick = () => Modal.Dismiss();
+            Modal.Show();
+
+            $J("#es_export_modal input[placeholder]").each(function() {
+                $J(this).attr('size', $J(this).attr('placeholder').length);
+            });
+
+            // attach the deferred's events to the modal
+            deferred.promise(Modal);
+        }`);
+
+        document.addEventListener("click", clickListener);
+
+        function clickListener(e) {
+            if (e.target.closest(".es_export_modal_submit")) {
+                e.preventDefault();
+                exportWishlist();
+                ExtensionLayer.runInPageContext(() => CModal.DismissActiveModal());
+            } else if (e.target.closest(".es_export_modal_close")) {
+                ExtensionLayer.runInPageContext(() => CModal.DismissActiveModal());
+            } else {
+                return;
+            }
+            document.removeEventListener("click", clickListener);
+        }
+
+        function exportWishlist() {
+            ExtensionLayer.runInPageContext(`function(){
+                let data = { rgAllApps: g_Wishlist.rgAllApps, g_rgAppInfo, options: {} };
```

Comment:
`g_Wishlist.rgAllApps` will be `undefined` if the wishlist data hasn't loaded yet.
From what I've seen by quickly skimming over wishlist.js, in line 128 there is the relevant Ajax request that, when done, initializes the variable.
It is possible to listen for this completion, iirc there should also be an example in our codebase

Reply:
I've put guards, that makes the export wishlist button do nothing until the data is loaded.

Reply:
If you mean the checks against `data` and co. this will just result in not downloading / copying anything at all without any graphical feedback.

Reply:
I see what you mean now. I'd have no clue where to look for this example. Can you point me to the right direction?

Reply:
`Ajax.Responders.register({ onComplete: () => Messenger.postMessage("ajaxCompleted") });`
That's the callback I meant, it has to registered in the page context